persistence-modules/spring-jdbc/src/main/java/com/baeldung/spring/jdbc/autogenkey/repository/MessageRepositoryJDBCTemplate.java

```diff
@@ -21,7 +22,8 @@ public long insert(final String message) {
         KeyHolder keyHolder = new GeneratedKeyHolder();
 
         jdbcTemplate.update(connection -> {

```

Comment:
Is Statement.RETURN_GENERATED_KEYS needed?

Reply:
Yes without it the test fails for me